libopenage/pathfinding/cost_field.h

```diff
@@ -95,15 +107,25 @@ class CostField {
 	 * Set the cost field values.
 	 *
 	 * @param cells Cost field values.
+	 * @param changed Time cost is set.
 	 */
-	void set_costs(std::vector<cost_t> &&cells);
+	void set_costs(std::vector<cost_t> &&cells, time::time_t &changed);
+
+	/**
+	 * Check if the cost field is dirty at the specified time.
+	 * 
+	 * @param time Specified time to check.
+	 */
```

Comment:
I am marking this as `Cost field is dirty if the cost field is accessed before the latest change to the cost field.`, but if we want to change this function to operate from the caller's POV, it will read:
`Cost field is dirty if the cost field is being accessed after the latest change to the cost field.`

Reply:
Probably related to my comment in https://github.com/SFTtech/openage/pull/1725#discussion_r1874183595